mithril.test.js

```diff
@@ -0,0 +1,33 @@
+import 'jest-dom/extend-expect'
+import m from 'mithril'
+import {getQueriesForElement, fireEvent, wait} from 'dom-testing-library'
+
+const Counter = () => {
+  let count = 0
+  return {
+    view: () =>
+      m(
+        'button',
+        {
+          onclick: () => {
+            count++
+          },
+        },
+        count,
+      ),
+  }
+}
+
+// tests:
+test('counter increments', async () => {
+  const div = document.createElement('div')
+  m.mount(div, Counter)
+
+  const {getByText} = getQueriesForElement(div)
+  const counter = getByText('0')
+  fireEvent.click(counter)
+  await wait(() => expect(counter).toHaveTextContent('1'))
```

Comment:
Instead of using `wait` here, could you use `fireEventAsync` (see `hyperapp.test.js` for an example).

Reply:
That's what I tried to begin with, but the test suite still fails. Here's the code:

```javascript
test('renders a counter', async () => {
  const div = document.createElement('div')
  m.mount(div, Counter)

  const {getByText} = getQueriesForElement(div)
  const counter = getByText('0')
  await fireEventAsync.click(counter)
  expect(counter).toHaveTextContent('1')

  await fireEventAsync.click(counter)
  expect(counter).toHaveTextContent('2')
})
```

Reply:
Weird. Maybe it takes longer than a single tick of the event loop to rerender for some reason. Ok, thanks!